genesyscloud/outbound_callabletimeset/resource_genesyscloud_outbound_callabletimeset.go

```diff
@@ -0,0 +1,139 @@
+	if err != nil {
+		return diag.Errorf("Failed to create Outbound Callabletimeset %s: %s", name, err)
+	}
+
+	d.SetId(*outboundCallabletimeset.Id)
+
+	log.Printf("Created Outbound Callabletimeset %s %s", name, *outboundCallabletimeset.Id)
+	return readOutboundCallabletimeset(ctx, d, meta)
+}
+
+// updateOutboundCallabletimeset is used by the outbound_callabletimeset resource to update an Outbound Callable Timeset
+func updateOutboundCallabletimeset(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := newOutboundCallableTimesetProxy(sdkConfig)
+
+	callableTimeset := getOutboundCallableTimesetFromResourceData(d)
+
+	log.Printf("Updating Outbound Callabletimeset %s", d.Id())
+	outboundCallabletimeset, err := proxy.updateOutboundCallabletimeset(ctx, d.Id(), &callableTimeset)
+	if err != nil {
+		return diag.Errorf("Failed to read Outbound Callabletimeset %s: %s", d.Id(), err)
+	}
+
+	log.Printf("Updated Outbound Callabletimeset %s", outboundCallabletimeset.Id)
+	return readOutboundCallabletimeset(ctx, d, meta)
+}
+
+// readOutboundCallabletimeset is used by the outbound_callabletimeset resource to read an Outbound Callable Timeset from the genesys cloud
+func readOutboundCallabletimeset(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := newOutboundCallableTimesetProxy(sdkConfig)
+
+	log.Printf("Reading Outbound Callabletimeset %s", d.Id())
+
+	return gcloud.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		callableTimeset, resp, getErr := proxy.getOutboundCallabletimesetById(ctx, d.Id())
+		if getErr != nil {
+			if gcloud.IsStatus404ByInt(resp) {
+				return retry.RetryableError(fmt.Errorf("Failed to read Outbound Callabletimeset %s: %s", d.Id(), getErr))
+			}
+			return retry.NonRetryableError(fmt.Errorf("Failed to read Outbound Callabletimeset %s: %s", d.Id(), getErr))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceOutboundCallabletimeset())
+
+		resourcedata.SetNillableValue(d, "name", callableTimeset.Name)
+		if callableTimeset.CallableTimes != nil {
+			// Remove the milliseconds added to start_time and stop_time by the API
+			trimTime(callableTimeset.CallableTimes)
+			d.Set("callable_times", flattenCallableTimes(*callableTimeset.CallableTimes))
+		}
+
+		log.Printf("Read Outbound Callabletimeset %s %s", d.Id(), *callableTimeset.Name)
+		return cc.CheckState()
+	})
+}
+
+// deleteOutboundCallabletimeset is used by the outbound_callabletimeset resource to delete an existing Outbound Callable Timeset from the genesys cloud
+func deleteOutboundCallabletimeset(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := newOutboundCallableTimesetProxy(sdkConfig)
```

Comment:
Use getOutboundCallabletimesetProxy instead of the newOutboundCallableTimesetProxy() function